Microsoft.Azure.CosmosRepository/src/Providers/DefaultCosmosContainerProvider.cs

```diff
@@ -58,13 +58,19 @@ async Task<Container> GetContainerValueFactoryAsync()
                         ? itemOptions.ContainerName
                         : _options.ContainerId,
                     PartitionKeyPath = itemOptions.PartitionKeyPath,
-                    UniqueKeyPolicy = itemOptions.UniqueKeyPolicy ?? new()
+                    UniqueKeyPolicy = itemOptions.UniqueKeyPolicy ?? new(),
+                    DefaultTimeToLive = itemOptions.DefaultTimeToLive
                 };
 
                 Container container =
                     await database.CreateContainerIfNotExistsAsync(
                         containerProperties).ConfigureAwait(false);
 
+                if (itemOptions.SyncContainerProperties)

```

Comment:
So this bit has to be flipped with each deployed to avoid the call to `ReplaceContainerAsync`? Is this call idempotent, or like a noop? I'm curious about the overhead of this, if it is not truly needed.

Reply:
Do you mean if two `IItem`s share a container then we do not need to replace the container more than once?

Reply:
Yes, if the app starts and does a replacement of the container - an expensive network I/O call. Then the app shuts down happily, and the next time it starts does it need to again run this same code? Or is it good now? My concern is that when this is set to `true` we're always adding I/O overhead to the startup time, which may not be necessary. I'm not sure if that is the case or not.

Reply:
Yes so if a client opts in to this then every time the ` Lazy<Task<Container>>` is resolved for the first time then yes it would make this `I/O` call. I have also added a service for this so a client can manually replace the container if needs be so they get to choose whether to pay this price for the I/O call or not.

Suppose with the service they could do this as part of a health check or something?

Reply:
This happens on the first request that involves a request to cosmos, not to the startup time of the application.

Reply:
If anything we just need to make that clear to consumers, so that they fully understand the potential ramifications of opting into it. It's almost like an EF migration.

Reply:
Yes, I agree it opted out by default so they would have to explicitly turn it on. Some docs on this would be good maybe even on the readme.